src/pages/CourseDetails/CourseDetails.jsx

```diff
@@ -24,49 +26,86 @@ const course = {
 };
 
 const CourseDetails = () => {
-  return (
-    <main className="content-width">
-      <h3 className="section-heading gutter-bottom-16 text-center">
-        Course Details
-      </h3>
-      <div className="course-details-wrapper">
-        <section>
-          <div className="img-wrapper gutter-bottom-8">
-            <img
-              className="responsive-img"
-              src={course.img}
-              alt={course.title}
-            />
-          </div>
-          <div className="course-content-details">
-            <div className="bubble">
-              <span>Content</span>
-              <span className="h5">{course.duration}</span>
+  const { productId } = useParams();
+  const [product, setProduct] = useState(null);
+  const [message, setMessage] = useState("loading....");
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(`/api/products/${productId}`);
```

Comment:
Everything is good and readable.

If you can destruct the response here use it .`

const {data:{product}} = fetch Call`